SimulationIOFilters/ImportDelamData.cpp

```diff
@@ -0,0 +1,568 @@
+      if(!ok)
+      {
+        QString ss = QObject::tr("Cannot read total CPT from Bvid StdOut File.");
+        setErrorCondition(-2007, ss);
+        return;
+      }
+      m_TotalThk = total_thk;
+    }
+    else if(tokens.size() >= 4 && tokens[3] == "plies")
+    {
+      bool ok = false;
+      int num_plies = tokens.last().toInt(&ok);
+      if(!ok)
+      {
+        QString ss = QObject::tr("Cannot read total number of plies from Bvid StdOut File.");
+        setErrorCondition(-2008, ss);
+        return;
+      }
+      m_NumPlies = num_plies;
+    }
+    else if(tokens.size() >= 4 && tokens[3] == "CZM")
+    {
+      bool ok = false;
+      int czmLayers = tokens.last().toInt(&ok);
+      if(!ok)
+      {
+        QString ss = QObject::tr("Cannot read total number of CZM layers from Bvid StdOut File.");
+        setErrorCondition(-2009, ss);
+        return;
+      }
+      m_CzmLayers = czmLayers;
+    }
+
+    bvidLine = bvidStdOutFile.readLine();
+  }
+
+  if(!m_Ex.has_value())
+  {
+    QString ss = QObject::tr("Cannot find Elx value from Bvid StdOut File.");
+    setErrorCondition(-2010, ss);
+    return;
+  }
+  if(!m_Lcx.has_value())
+  {
+    QString ss = QObject::tr("Cannot find Lcx value from Bvid StdOut File.");
+    setErrorCondition(-2011, ss);
+    return;
+  }
+  if(!m_TotalThk.has_value())
+  {
+    QString ss = QObject::tr("Cannot find total CPT from Bvid StdOut File.");
+    setErrorCondition(-2012, ss);
+    return;
+  }
+  if(!m_NumPlies.has_value())
+  {
+    QString ss = QObject::tr("Cannot find total number of plies from Bvid StdOut File.");
+    setErrorCondition(-2013, ss);
+    return;
+  }
+  if(!m_CzmLayers.has_value())
+  {
+    QString ss = QObject::tr("Cannot find total number of CZM layers from Bvid StdOut File.");
+    setErrorCondition(-2014, ss);
+    return;
+  }
+
+  bvidStdOutFile.close();
+}
+
+// -----------------------------------------------------------------------------
+size_t ImportDelamData::getBvidFileLineCount()
+{
+  QFile bvidFile(m_CSDGMFile);
+  if(!bvidFile.open(QFile::ReadOnly))
+  {
+    QString ss = QObject::tr("Cannot open Bvid File for reading.");
+    setErrorCondition(-2015, ss);
+    return 0;
+  }
+
+  size_t lineCount = 0;
+  while(!bvidFile.readLine().isEmpty())
+  {
+    lineCount++;
+  }
+
+  bvidFile.close();
+
+  return lineCount;
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::execute()
+{
+  initialize();
+  dataCheck();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  readCSDGMFile();
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::readCSDGMFile()
+{
+  QFile csdgmFile(m_CSDGMFile);
+  if(!csdgmFile.open(QFile::ReadOnly))
+  {
+    QString ss = QObject::tr("Cannot open Bvid File for reading.");
+    setErrorCondition(-2016, ss);
+    return;
+  }
+
+  QString csdgmLine = csdgmFile.readLine();
+  size_t lineCount = 1;
```

Comment:
Minor detail but this is a misleading variable name. Consider renaming it to lineNum or something more accurate.

The error messages do not make sense when outputting the total line count instead of the current line number.